EVM documentation improvements

dssei opened the pull request:
Adding rust docs for EVM related stucts/functions

dssei commented:
@ARitz-Cracker please have a look.

codchen commented on packages/sei-cosmwasm/src/querier.rs:
```diff
@@ -187,7 +206,19 @@ impl<'a> SeiQuerier<'a> {
+    /// Query to get hex payload for the ERC-20 `transfer` function
+    ///
+    /// # Arguments
+    /// * `recipient` - Sei recipient address.
```
cosmos

dssei replied:
Yeah, I guess through the docs I was referring to x0 addresses as `EVM` and to bech32 as `Sei`. But Sei is probably confusing as we support both.

ARitz-Cracker reviewed (changes requested):
Very much appreciated! Though I still wholly believe that more explicit language would benefit newcomers. Most of my comments can apply to other sections of this PR.

dssei commented on packages/sei-cosmwasm/src/msg.rs:
```diff
@@ -41,13 +41,25 @@ pub enum SeiMsg {
     },
+    /// Calls EVM contract deployed at `to` address with specified `value` and `data`.
+    /// The from address is the contract address of the contract executing the call.
+    /// The EVM (Solidity) contract `msg.sender` in this case will be the CW contract address.
```
Done

dssei commented on packages/sei-cosmwasm/src/msg.rs:
```diff
@@ -41,13 +41,25 @@ pub enum SeiMsg {
         metadata: Metadata,
     },
+    /// Calls EVM contract deployed  at `to` address with the given `data`.
+    /// The from address is the callers address.
```
Replaced with suggested text

dssei commented:
> Very much appreciated! Though I still wholly believe that more explicit language would benefit newcomers. Most of my comments can apply to other sections of this PR.

@ARitz-Cracker thanks for reviewing. Will be addressing most comments shortly

codchen commented on packages/sei-cosmwasm/src/msg.rs:
```diff
@@ -41,13 +41,25 @@ pub enum SeiMsg {
+        /// The address of the EVM contract to call
         to: String,
+        /// Base64 encoded data to pass to the contract
         data: String, // base64 encoded
```
yes all bytes will be transferred in base64 format

codchen commented on packages/sei-cosmwasm/src/msg.rs:
```diff
@@ -41,13 +41,25 @@ pub enum SeiMsg {
     },
+    /// Calls EVM contract deployed  at `to` address with the given `data`.
+    /// The from address is the callers address.
+    /// Please note that the CW contract has to be in allow list in order to execute delegate call.
```
it refers to any CW code hash in this list https://github.com/sei-protocol/sei-chain/blob/seiv2/x/evm/types/params.go#L142 (currently only the pointer contracts).

codchen commented on packages/sei-cosmwasm/src/msg.rs:
```diff
@@ -41,13 +41,25 @@ pub enum SeiMsg {
+    /// The EVM (Solidity) contract `msg.sender` in this case will be the callers address.
     DelegateCallEvm {
+        /// The address of the EVM contract to call
         to: String,
```
no it doesn't

codchen commented on packages/sei-cosmwasm/src/msg.rs:
```diff
@@ -41,13 +41,25 @@ pub enum SeiMsg {
+    /// Calls EVM contract deployed  at `to` address with the given `data`.
+    /// The from address is the callers address.
+    /// Please note that the CW contract has to be in allow list in order to execute delegate call.
+    /// The EVM (Solidity) contract `msg.sender` in this case will be the callers address.
```
@besated do you know if we already have a sei doc page for address mapping? If not I can help with writing up one (not sure where the gitbook repo is)

besated replied:
we have this section in the docs - https://v2.docs.sei.io/setting-up-a-wallet#advanced

if you want to add more details, you can create a pr in the `sei-docs` repo

dssei commented:
@ARitz-Cracker addressed the most comments and will merge it. We can add more improvements in subsequent PRs
